src/PrpShop/QPlasmaTreeItem.cpp

```diff
@@ -15,88 +15,163 @@
     : QTreeWidgetItem(parent, kTypeNone)
 {
-    setIcon(0, QIcon(":/img/folder.png"));
+    reinit();
 }
 
 QPlasmaTreeItem::QPlasmaTreeItem(QTreeWidget* parent, plKey obj)
-    : QTreeWidgetItem(parent, kTypeKO), fObjKey(obj)
+    : QTreeWidgetItem(parent, kTypeKO), fObjKey(std::move(obj))
+{
+    reinit();
+}
+
+QPlasmaTreeItem::QPlasmaTreeItem(QTreeWidget* parent, short classType)
+    : QTreeWidgetItem(parent, kTypeClassType), fClassType(classType)
 {
-    setText(0, st2qstr(obj->getName()));
-    setIcon(0, pqGetTypeIcon(obj->getType()));
+    reinit();
 }
 
 QPlasmaTreeItem::QPlasmaTreeItem(QTreeWidget* parent, const QString& age)
     : QTreeWidgetItem(parent, kTypeAge), fHasBuiltIn(false),
       fHasTextures(false), fAge(age)
 {
-    setText(0, age);
-    setIcon(0, QIcon(":/img/age.png"));
+    reinit();
 }
 
 QPlasmaTreeItem::QPlasmaTreeItem(QTreeWidget* parent, plPageInfo* page)
     : QTreeWidgetItem(parent, kTypePage), fPage(page)
 {
-    setText(0, st2qstr(page->getPage()));
-    setIcon(0, QIcon(":/img/page.png"));
+    reinit();
 }
 
 QPlasmaTreeItem::QPlasmaTreeItem(QTreeWidgetItem* parent)
     : QTreeWidgetItem(parent, kTypeNone)
 {
-    setIcon(0, QIcon(":/img/folder.png"));
+    reinit();
 }
 
 QPlasmaTreeItem::QPlasmaTreeItem(QTreeWidgetItem* parent, plKey obj)
-    : QTreeWidgetItem(parent, kTypeKO), fObjKey(obj)
+    : QTreeWidgetItem(parent, kTypeKO), fObjKey(std::move(obj))
+{
+    reinit();
+}
+
+QPlasmaTreeItem::QPlasmaTreeItem(QTreeWidgetItem* parent, short classType)
+    : QTreeWidgetItem(parent, kTypeClassType), fClassType(classType)
 {
-    setText(0, st2qstr(obj->getName()));
-    setIcon(0, pqGetTypeIcon(obj->getType()));
+    reinit();
 }
 
 QPlasmaTreeItem::QPlasmaTreeItem(QTreeWidgetItem* parent, const QString& age)
     : QTreeWidgetItem(parent, kTypeAge), fHasBuiltIn(false),
       fHasTextures(false), fAge(age)
 {
-    setText(0, age);
-    setIcon(0, QIcon(":/img/age.png"));
+    reinit();
 }
 
 QPlasmaTreeItem::QPlasmaTreeItem(QTreeWidgetItem* parent, plPageInfo* page)
     : QTreeWidgetItem(parent, kTypePage), fPage(page)
 {
-    setText(0, st2qstr(page->getPage()));
-    setIcon(0, QIcon(":/img/page.png"));
+    reinit();
+}
+
+static QString pqFormatPageName(const plPageInfo* page)
+{
+    if (s_showAgePageIDs) {
+        return st2qstr(ST::format("{} {}", page->getLocation().toString(), page->getPage()));
+    } else {
+        return st2qstr(page->getPage());
+    }
+}
+
+void QPlasmaTreeItem::reinit()
+{
+    switch (type()) {
+        case kTypeNone:
+            setIcon(0, QIcon(":/img/folder.png"));
+            break;
+
+        case kTypeKO:
+            setText(0, st2qstr(fObjKey->getName()));
+            setIcon(0, pqGetTypeIcon(fObjKey->getType()));
+            break;
+
+        case kTypeClassType:
+            setText(0, pqGetFriendlyClassName(fClassType));
+            setIcon(0, QIcon(":/img/folder.png"));
+            break;
+
+        case kTypeAge:
+            setText(0, fAge);
```

Comment:
Would we want to show the sequence prefix next to Age folder names too? It's probably rare that people are opening multiple Ages in PlasmaShop, but the wording of "Show Age/Page IDs" makes me expect it would also be displayed at the Age level

Reply:
Yeah, that would make sense. I think I was too lazy to implement that part because the age items currently only store the name and not the sequence prefix... But it shouldn't be hard to add the extra field for that.

Reply:
Done:

![Screenshot of Object Browser with age IDs visible next to both age and page items](https://github.com/H-uru/PlasmaShop/assets/6641959/f4a3bac7-89e6-4474-bb39-71dee118bc90)